src/iuse_actor.cpp

```diff
@@ -3104,3 +3105,41 @@ long place_trap_actor::use( player * const p, item * const it, bool, const tripo
     }
     return 1;
 }
+
+void emit_actor::load( JsonObject &obj )
+{
+    assign( obj, "emits", emits );
+    assign( obj, "scale_qty", scale_qty );
```

Comment:
See below. Express your `emit` definitions as per-unit volume and then provide a boolean `scaled` field on the actor to enable scaling with charges.